src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -236,20 +199,55 @@ public Task WriteEndBatchAsync()
         }
 #endif
 
-        /// <summary>Starts a new changeset; can only be called after WriteStartBatch and if no other active operation or changeset exists.</summary>
+        /// <summary>
+        /// Starts a new changeset without specifying group id.
+        /// This can only be called after WriteStartBatch and if no other active operation or changeset exists.
+        /// </summary>
         public void WriteStartChangeset()
+        {
+            this.WriteStartChangeset(null);
+        }
+
+        /// <summary>
+        /// Starts a new changeset with the specified group id, which is null for multipart batch.
+        /// This can only be called after WriteStartBatch and if no other active operation or changeset exists.</summary>
+        /// <param name="groupId">
+        /// For Json batch, this is the atomic group id. Default value of null can be used for writing request and an GUID will be generated.
+        /// Non-null value from corresponding request atomic group should be used for writing response so that atomic groups
+        /// from request and response can be correlated.
+        /// For multipart batch, null value should be used and is ignored.
+        /// </param>
+        public void WriteStartChangeset(string groupId)

```

Comment:
>string groupId [](start = 40, length = 14)

disallow passing null, so that downstream logic can rely  on a non-null groupid (either passed by user or a guid generated in the parameterless WriteStartChangeset).

Reply:
updated to throw exception if it is null.

---
In reply to: [155383773](https://github.com/OData/odata.net/pull/992#discussion_r155383773) [](ancestors = 155383773)

Reply:
You can just use:
             ExceptionUtils.CheckArgumentNotNull(groupOrChangesetId, "groupOrChangesetId");


---
In reply to: [155405601](https://github.com/OData/odata.net/pull/992#discussion_r155405601) [](ancestors = 155405601,155383773)

Reply:
updated. Also updated test as well.

---
In reply to: [156767675](https://github.com/OData/odata.net/pull/992#discussion_r156767675) [](ancestors = 156767675,155405601,155383773)